packages/plugin-react/src/index.ts

```diff
@@ -211,11 +216,8 @@ export default function viteReact(opts: Options = {}): PluginOption[] {
         filepath.match(fileExtensionRE) ||
         []
 
-      if (/\.(?:mjs|[tj]sx?)$/.test(extension)) {
+      if (filter(id)) {
```

Comment:
I think this changes behavior when `id` is `foo.txt?a=.jsx`. (it was `true` but now it's `false`)